### PR TITLE
BOAC-326, 'inIntensiveCohort' is new attribute in cohortFilters object and checkbox on cohort view

### PR DESCRIPTION
**Pull request by johncrossman**
https://jira.ets.berkeley.edu/jira/browse/BOAC-326

UX is in the Jira. I ripped out `/api/intensive_cohort` endpoint in favor of optional 'intensive' arg in `/api/student`. A remaining task: the manageMyCohorts view must include proper "pill" icon if intensive is true.

### CODE REVIEW

File: boac/models/cohort_filter.py [hunk 1]
```diff
@@ -43,13 +43,17 @@ def create(
             levels=None,
             majors=None,
             unit_ranges=None,
+            in_intensive_cohort=None,
     ):
+        # If in_intensive_cohort is True then search intensive cohort; if equals False then search
+        # non-intensive students; if equals None then search all students.
```
Comment (johncrossman) [inline boac/models/cohort_filter.py]: The front-end plays by the rules described above. IMO, having this standard (True, False or None) on the backend might be beneficial for future use cases (eg, search for `is_inactive`).

File: boac/static/app/cohort/cohortController.js [hunk 2]
```diff
@@ -519,6 +542,13 @@
       }
     };
 
+    $scope.disableSearchButton = function() {
+      // Disable button if page is loading or no search criterion is selected
+      var count = $scope.search.count;
+      return $scope.isLoading || $scope.isSaving || !count.gpaRanges && !count.groupCodes && !count.levels &&
+        !count.majors && !count.unitRanges && ($scope.showIntensiveCheckbox && !$scope.search.options.intensive);
```
Comment (paulkerschen) [inline boac/static/app/cohort/cohortController.js]: @johncrossman Would more parentheses make precedence on these booleans clearer? I always get lost when both `&&` and `||` are involved.

Comment (johncrossman) [reply boac/static/app/cohort/cohortController.js]: @paulkerschen Yes, but our linter disagrees. I can disable the rule and fix. Opinion?

Comment (johncrossman) [reply boac/static/app/cohort/cohortController.js]:
```
  548:53  error  Gratuitous parentheses around expression  no-extra-parens
```

Comment (johncrossman) [reply boac/static/app/cohort/cohortController.js]: The `no-extra-parens` rule is off. Let the entropy begin.

File: boac/api/cohort_controller.py [hunk 3]
```diff
@@ -92,6 +75,7 @@ def create_cohort():
     levels = util.get(params, 'levels')
     majors = util.get(params, 'majors')
     unit_ranges = util.get(params, 'unitRanges')
+    in_intensive_cohort = bool(util.get(params, 'intensive'))
```
Comment (johncrossman) [inline boac/api/cohort_controller.py]: I just noticed that the line above needs `to_bool_or_none` instead of `bool()`.  Amend coming in moments.

Comment (paulkerschen) [reply boac/api/cohort_controller.py]: @johncrossman Looks like a test might need to be updated to match the change?

Comment (johncrossman) [reply boac/api/cohort_controller.py]: @paulkerschen  Ah yes

Comment (johncrossman) [reply boac/api/cohort_controller.py]: Fyi, fix coming in ~10 min.

**Comment by johncrossman** [comment]
Fyi, Travis green.  cc: @paulkerschen

**Comment by paulkerschen** [comment]
And BOAC goes ternary. Thanks @johncrossman.

**Pull request merged**